src/test/java/de/siegmar/logbackgelf/custom/CustomGelfEncoder.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Logback GELF - zero dependencies Logback GELF appender library.
+ * Copyright (C) 2016 Oliver Siegmar
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package de.siegmar.logbackgelf.custom;
+
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+
+import de.siegmar.logbackgelf.GelfEncoder;
+import de.siegmar.logbackgelf.GelfMessage;
+
+// Put it in different package from GelfEncoder to reveal any visibility issues
+public class CustomGelfEncoder extends GelfEncoder {
+
+    private static final int INT_255 = 0xff;
+    private static final char CHAR_ZERO = '0';
+
+    @Override
+    protected String gelfMessageToJson(final GelfMessage gelfMessage) {
+        final String json = super.gelfMessageToJson(gelfMessage);
+        final String sha256 = sha256(json);
+        gelfMessage.getAdditionalFields().put("sha256", sha256);
+        return super.gelfMessageToJson(gelfMessage);
+    }
+
+    // Based on https://stackoverflow.com/a/11009612/74694
+    private static String sha256(final String base) {
```

Comment:
As it is a test class, please keep using your initial Guava implementation for the sake of readability / maintainability.

Reply:
How do I explain that to checkstyle?

> Disallowed import - com.google.common.hash.Hashing. [ImportControl]

Reply:
You can exclude the `ImportControl` check in `config/checkstyle/suppressions.xml` for that class. Classes that match the regular expression `.*Test.java` are already excluded.

Reply:
Or should I add that to the list of allowed imports?

Reply:
Unfortunately I couldn't find a clean solution to provide separate import files for productional and test code. So, adding this to the allowed imports would allow a Guava import in the production code. Right now, it wouldn't be a problem because the gradle configuration only adds the dependency for the tests, but that could be confusing in the future...

Reply:
Fixed